alembic/versions/12c0ccf6b825_create_trigger.py

```diff
@@ -0,0 +1,98 @@
+"""create trigger
+
+Revision ID: 12c0ccf6b825
+Revises: c8ef88c250e7
+Create Date: 2024-10-27 20:27:30.805137
+
+"""
+from alembic import op
+import sqlalchemy as sa
+
+
+# revision identifiers, used by Alembic.
+revision = '12c0ccf6b825'
+down_revision = 'c8ef88c250e7'
+branch_labels = None
+depends_on = None
+
+create_trigger = """CREATE OR REPLACE FUNCTION log_site_changes()
+RETURNS TRIGGER AS $$
+DECLARE
+    user_uuid UUID;
+BEGIN
+    user_uuid := nullif(current_setting('pvsite_datamodel.current_user_uuid', true), '')::UUID;
+    
+    IF TG_OP = 'INSERT' THEN
+        INSERT INTO h_sites (
+            site_history_uuid,
+            site_uuid,
+            site_data,
+            changed_by,
+            operation_type,
+            created_utc
+        ) VALUES (
+            gen_random_uuid(),
+            NEW.site_uuid,
+            to_jsonb(NEW),
+            user_uuid,
+            'INSERT',
+            NOW()
+        );
+
+    ELSIF TG_OP = 'UPDATE' THEN
```

Comment:
coudl this first two cases come together, and the `operation_type` be `TG_OP`. I have to admit this is not my area of expertise though